results_spreadsheet.py

```diff
@@ -0,0 +1,136 @@
+
+    # Map either 'r' or 'ba' to state (whichever exists)
+    col = {"r", "ba"}.intersection(df.columns)
+    if col:
+        col = col.pop()
+        df["state"] = df[col].map(region_to_state)
+        df = df.drop(columns=[col])
+
+    # collapse i
+    if "i" in df.columns:
+        df["i"] = df["i"].astype(str).str.strip().str.lower()
+        for tech in agg_techs:
+            if tech in ["coal", "gas"]:
+                df.loc[
+                    df["i"].str.contains(tech, case=False, na=False)
+                    & ~df["i"].str.contains("ccs", case=False, na=False),
+                    "i",
+                ] = tech
+                df.loc[
+                    df["i"].str.contains(tech, case=False, na=False)
+                    & df["i"].str.contains("ccs", case=False, na=False),
+                    "i",
+                ] = f"{tech}_ccs"
+            else:
+                df.loc[df["i"].str.contains(tech, case=False, na=False), "i"] = tech
+
+    # add Scenario
+    df.insert(0, "scenario", scenario)
+
+    # aggregate value
+    # Identify numeric columns to aggregate
+    if "value" in df.columns:
+        agg_cols = ["value"]
+    elif {"tons", "md", "damage_$", "mortality"}.issubset(df.columns):
+        agg_cols = ["tons", "md", "damage_$", "mortality"]
+    else:
+        agg_cols = []
+
+    if agg_cols:
+        # Group by all non-aggregated columns
+        group_cols = [c for c in df.columns if c not in agg_cols]
+        df = df.groupby(group_cols, as_index=False)[agg_cols].sum()
+    else:
+        df = df.drop_duplicates()
+
+    return df
+
+
+# === Main ===
+summary_df = pd.read_csv(summary_file)
+region_to_state = load_region_map(mapping_file)
+csv_types = [c for c in summary_df.columns if c.lower() != "scenario"]
+
+tabs = {csv_name: [] for csv_name in csv_types}
+
+# Wrap scenario loop in tqdm progress bar
+for _, row in tqdm(
+    summary_df.iterrows(), total=len(summary_df), desc="Processing scenarios"
+):
+    scenario = row["Scenario"]
+    for csv_name in csv_types:
+        csv_path = os.path.join(base_path, scenario, csv_name)
+        if not os.path.exists(csv_path):
+            continue
+        try:
+            df = pd.read_csv(csv_path)
+            df = clean_and_aggregate(df, scenario, region_to_state)
+            tabs[csv_name].append(df)
+        except Exception as e:
+            print(f"Error processing {csv_path}: {e}")
+
+# === Write Excel ===
+total_rows = 0
+for dfs in tabs.values():
+    if dfs:
+        total_rows += sum(len(df) for df in dfs)
+print(f"\nTotal rows across all sheets: {total_rows:,}\n")
+
+if os.path.exists(output_excel):
+    # Must use openpyxl for append mode
+    excel_engine = "openpyxl"
+    mode = "a"
+    if_exists = "replace"
+    print(f"Updating existing file: {output_excel}")
+else:
+    # Prefer xlsxwriter for new files, fall back to openpyxl
+    try:
+        import xlsxwriter
+        excel_engine = "xlsxwriter"
```

Comment:
Did you have to install xlsxwriter to the `reeds-viz` environment? If so, can you add it to the `environment.yml` file?